taskchampion/taskchampion/src/server/sync/mod.rs

```diff
@@ -28,8 +29,16 @@ impl SyncServer {
     /// identify this client to the server.  Multiple replicas synchronizing the same task history
     /// should use the same client_id.
     pub fn new(origin: String, client_id: Uuid, encryption_secret: Vec<u8>) -> Result<SyncServer> {
+        let origin = Url::parse(&origin)
+            .map_err(|_| Error::Server(format!("Could not parse {} as a URL", origin)))?;
+        if origin.path() != "/" {
+            return Err(Error::Server(format!(
+                "Server origin must have an empty path; got {}",
+                origin
+            )));
```

Comment:
Why does the origin require an empty path? I hope this is sufficiently documented in taskchampion-sync-server so people don't set it up on a subdirectory before running into this error.

Reply:
An [origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin) is the scheme/hostname/port, so I think with the current naming of the config a path is prohibited. I'll file a new bug to rename to url and allow a path (with trivial backward compatibility)